addons/im_livechat/static/src/models/chatbot.js

```diff
@@ -6,12 +6,24 @@ import { attr, one } from '@mail/model/model_field';
 registerModel({
     name: 'Chatbot',
     identifyingFields: ['livechatButtonViewOwner'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeName() {
+            return this.data.name;
+        },
```

Comment:
In the PR that introduces `Chatbot/scriptId`, you guard `this.data`. Why do it in `_computeScriptId` but not in `_computeName`?

Reply:
why put the guard in some computes then?

Reply:
> why put the guard in some computes then?

You mean which compute exactly? Sometimes guard is necessary when the field being accessed in not defined.
I think I put the guard at some point because I mistakenly defined `LivechatButtonView/chatbot` as `default: insertAndReplace()` in an ealier version. Then I fixed with proper imperative update, and then declarative version that you have seen in final.

I might have forgotten to remove some guards, if there are some left.